doc/trajopt_planner/trajopt_planner_tutorial.rst

```diff
@@ -0,0 +1,123 @@
+TrajOpt Planner
+===============
+TrajOpt is a sequential convex optimization algorithm for motion planning problems where the non-convex, non-affine equality, and non-equality constraints are relaxed, approximately linearized and convexified to create an objective function. This work has been done based on [schulman2013]_ and the `original implementation <https://github.com/ros-industrial-consortium/trajopt_ros>`_. Sequential convex programming, simply refers to using a convex model and repeatedly minimizing it. Consider the following non-convex problem:
```

Comment:
I know I'm reviewing this PR a bit late, but I don't think this first paragraph is a good intro: a lot of technical jargon thrown out, and a lot of unnecessary details: all the users of the tutorial need to know is that it's an optimization algorithm, and can take costs. IMO, everything before Getting started isn't needed and just confuses the user.

For example, the [CHOMP tutorial](https://ros-planning.github.io/moveit_tutorials/doc/chomp_planner/chomp_planner_tutorial.html) handles this balance between too technical and not enough information well; it highlights the main differences between sampling and optimization planners, but doesn't go into the exact gradient descent equation or algorithm. You've cited the paper itself, which is good enough.

Reply:
I see. I guess it depends on who the audience are and at what level we want to explain the algorithm. I am open to shrink or even remove this section if it confuses the reader.